test/trip_dispatch_test.rb

```diff
@@ -78,8 +74,55 @@ def build_test_dispatcher
     end
   end
 
-  # TODO: un-skip for Wave 2
-  xdescribe "drivers" do
+  # Wave 3: tests for request_trip method
+  describe "request_trip" do
+    it "returns instance of trip" do
+      td = RideShare::TripDispatcher.new
+      expect(td.request_trip(1)).must_be_kind_of RideShare::Trip
+    end
+    # Were the trip lists for the driver and passenger updated?
+    it "Adds trip to passenger's trips" do
+      dispatcher = build_test_dispatcher
+      passenger = dispatcher.find_passenger(1)
+      dispatcher.request_trip(1)
+      expect(passenger.trips.length).must_equal 3
+    end
+
+    it "Adds trip to driver's trips" do
+      dispatcher = build_test_dispatcher
+      passenger = dispatcher.find_passenger(1)
+      dispatcher.request_trip(1)
+
+      # get the passenger's last trip
+      passengers_last_trip = passenger.trips[-1]
+      last_trips_driver = passengers_last_trip.driver_id
+      driver = dispatcher.find_driver(last_trips_driver)
+      expect(driver.trips.length).must_equal 5
+    end
+
+    it "Was the driver available" do
+      dispatcher = build_test_dispatcher
+      drivers = dispatcher.find_available_drivers
+      passenger = dispatcher.find_passenger(1)
+      dispatcher.request_trip(1)
+
+      passengers_last_trip = passenger.trips[-1]
+      last_trips_driver = passengers_last_trip.driver_id
+      driver = dispatcher.find_driver(last_trips_driver)
+      expect(drivers.include?(last_trips_driver)).must_equal false
```

Comment:
`last_trips_driver` is an ID... so it'll be 2 at this point. It won't be an instance of Driver. `drivers` is an array of instances of `Driver`s. It will never carry driver IDs by y'alls implementation. You all probably WANTED this line to be

```ruby
expect(drivers.include?(driver)).must_equal false
```

since `driver` is an instance of `Driver`, but it didn't pass

This test has the right intention-- it's checking that the list of available drivers no longer has the newly unavailable driver. But that's not how arrays of objects work! The `drivers` array will always have Driver 2, because it was built with the line `drivers = dispatcher.find_available_drivers` BEFORE Driver 2 got assigned a trip, so it includes Driver 2. There is no line of code that says "take out Driver 2 because Driver 2's status changed to unavailable."

If I were to refactor this line, I'd probably make the array of available drivers now, AFTER the Act step. I might do

```ruby
expect(dispatcher.find_available_drivers).wont_include passenger.trips.last.driver
```

The test line below (directly checking the status of Driver 2) should remain, too!